### DISCUSSION TITLE
Timeout limit not working on serverless functions

### DISCUSSION THREAD

**Discussion by EricHasegawa** [vercel/next.js#58605] [Help] [answered]
### Summary

Despite adding `export const maxDuration = 300` and `export const dynamic = 'force-dynamic`, and trying to add a vercel.json with the following

```
{
    "functions": {
        "pages/api/**": {
            "memory": 3008,
            "maxDuration": 300
        },
    }
}
```

I can't get my timeout limits to go above 60s for my API endpoints. I'm running NextJs version 13.5.6, am using the `pages` router, and am on a pro plan, and Node v18, have looked over the docs thoroughly and ran through relevant threads. Fairly stumped on what to try next.

Related [Stack Overflow](https://stackoverflow.com/questions/77503770/how-to-increase-timeout-limit-on-vercel-serverless-functions) post

**Comment by icyJoseph** [comment]
- https://vercel.com/docs/functions/serverless-functions/runtimes#size-limits

Are you in a Hobby plan? Then you can only have 10 seconds max.

**Reply by EricHasegawa** [reply]
No I am on a pro plan - using many pro plan features, am VERY certain that I am not on a hobby plan

**Reply by icyJoseph** [reply, marked as answer]
Alright, and, do yo have a `src` folder? Like `src/pages/api/foo.js`? I think the `src` path has to be part of the vercel.json
